lucene/spatial-extras/src/java/org/apache/lucene/spatial/prefix/tree/DateRangePrefixTree.java

```diff
@@ -500,4 +516,18 @@ public Calendar parseCalendar(String str) throws ParseException {
     throw new ParseException("Improperly formatted date: "+str, offset);
   }
 
+  private  void isValidDateDelimeter(String str, int offset, char delim) {
+    if(str.charAt(offset) != delim) {
+      throw new IllegalArgumentException("Not the valid delimeter for Position"+offset);
+    }
+  }
+
+  private int parseIntegerAndValidate(String  str, int offset, int min, int max) {
+    int val = Integer.parseInt(str.substring(offset, offset+2));
+    if((val < min ) || (val>max)) {
+      throw new IllegalArgumentException("Not valid date.");
```

Comment:
I suppose expected and actual character might be reported here, assuming enclosing ParseException report whole input string and the position.  The same is applicable to IAE above.

Reply:
I have incorporated the same.

Reply:
@gd-spb-e1m I have incorporated the same, please review